### PR TITLE
minor sonar bug analysis changes

### PR DESCRIPTION
**Pull request by lazerion**
According to https://sonarcloud.io/project/issues?id=com.hazelcast%3Ahazelcast-root&resolved=false&severities=BLOCKER&statuses=CONFIRMED&types=BUG

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java [hunk 1]
```diff
@@ -515,16 +515,24 @@ private void handleAt(String first) {
...
                 String l = br.readLine();
                 while (l != null) {
                     handleCommand(l);
                     l = br.readLine();
                 }
-                br.close();
             } catch (IOException e) {
                 e.printStackTrace();
+            } finally {
+                if (br != null) {
```
Comment (kwart) [inline hazelcast-client/src/main/java/com/hazelcast/client/console/ClientConsoleApp.java]: We have `IOUtil.closeResource()` for these cases.

File: hazelcast/src/main/java/com/hazelcast/console/ConsoleApp.java [hunk 2]
```diff
@@ -485,16 +485,24 @@ private void handleAt(String first) {
...
                 String l = br.readLine();
                 while (l != null) {
                     handleCommand(l);
                     l = br.readLine();
                 }
-                br.close();
             } catch (IOException e) {
                 e.printStackTrace();
+            } finally {
+                if (br != null) {
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/console/ConsoleApp.java]: dtto - `IOUtil.closeResource()`

File: hazelcast/src/main/java/com/hazelcast/osgi/impl/OSGiScriptEngineManager.java [hunk 3]
```diff
@@ -309,16 +309,22 @@ private ScriptEngineManagerInfo createScriptEngineManagerInfo(String factoryName
...
+                            new InputStreamReader(u.openStream(), "UTF-8"));
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        line = line.trim();
+                        if (!line.startsWith("#") && line.length() > 0) {
+                            factoryCandidates.add(line);
+                        }
+                    }
+                } finally {
+                    if (reader != null) {
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/osgi/impl/OSGiScriptEngineManager.java]: I would use the `IOUtil.closeResource()` here too.

**Comment by Holmistr** [comment]
The PR builder has really a loooooot of failures :-/ @lazerion you should take a look.

**Comment by lazerion** [comment]
run-lab-run

**Comment by lazerion** [comment]
Thanks @Holmistr , all checks now passing, therefore merging

**Pull request merged**